src/librustc/ty/instance.rs

```diff
@@ -369,6 +369,24 @@ impl<'tcx> Instance<'tcx> {
         Instance { def, substs }
     }
 
+    /// Returns the substitutions that should be used on the MIR body for this `Instance`, if at
+    /// all.
```

Comment:
I'd like to see this comment expanded a bit. I think the scenario is pretty confusing. Can we steal some of the text from https://github.com/rust-lang/rust/issues/69925 to explain what's going on?

Perhaps something like this:

---

Depending on the kind of `InstanceDef`, the MIR body associated with an instance is expressed in terms of the generic parameters of `self.def_id()`, and in other cases the MIR body is expressed in terms of the types found in the substitution array. In the former case, we want to substitute those generic types and replace them with the values from the `substs` when monomorphizing the function body. But in the latter case, we don't want to do that substitution, since it has already been done effectively. This function returns `Some(substs)` in the former case and `None` otherwise -- i.e., if this function returns `None`, then the MIR body does not require substitution during monomorphization.

Reply:
Probably add a `FIXME(#69925)` as well?

Reply:
Fixed.